lib/xenops_sandbox.ml

```diff
@@ -62,15 +62,16 @@ end = struct
   let qemu_base_uid () = (Unix.getpwnam "qemu_base").Unix.pw_uid
   let qemu_base_gid () = (Unix.getpwnam "qemu_base").Unix.pw_gid
 
-  let of_domid ~daemon ~domid =
-    let root = Printf.sprintf "/var/run/xen/%s-root-%d" daemon domid in
+  let of_domid ~daemon ~domid ~vm_uuid =
+    let root = if domid = 0 then Printf.sprintf "/var/run/xen/%s-root-%d-%s" daemon domid vm_uuid
+      else Printf.sprintf "/var/run/xen/%s-root-%d" daemon domid in
```

Comment:
Good, we want to preserve the old chroot name for VMs when you update Naples->Plymouth

Reply:
Wouldn't they have a new domid after an upgrade though?